django/contrib/postgres/operations.py

```diff
@@ -256,3 +258,63 @@ def describe(self):
     @property
     def migration_name_fragment(self):
         return 'remove_collation_%s' % self.name.lower()
+
+
+class AddConstraintNotValid(AddConstraint):
+    """
+    Add a table constraint without enforcing validation, using PostgreSQL's
+    NOT VALID syntax.
+    """
+    def __init__(self, model_name, constraint):
+        if not isinstance(constraint, CheckConstraint):
+            raise NotSupportedError('AddConstraintNotValid does not support %s.' % (
+                constraint.__class__.__name__))
+        super().__init__(model_name, constraint)
+
+    def describe(self):
+        return 'Create NOT VALID constraint %s on model %s' % (
+            self.constraint.name, self.model_name
+        )
+
+    def database_forwards(self, app_label, schema_editor, from_state, to_state):
+        model = from_state.apps.get_model(app_label, self.model_name)
+        if self.allow_migrate_model(schema_editor.connection.alias, model):
+            constraint_sql = self.constraint.create_sql(model, schema_editor)
+            if constraint_sql:
+                schema_editor.execute(str(constraint_sql) + " " + "NOT VALID")
+
+    @property
+    def migration_name_fragment(self):
+        return super().migration_name_fragment + '_not_valid'
+
+
+class ValidateConstraint(IndexOperation):
+    """
+    Validate a table constraint on existing rows using PostgreSQL's VALID CONSTRAINT syntax.
+    """
+    option_name = 'constraints'
+
+    def __init__(self, model_name, name):
+        self.model_name = model_name
+        self.constraint_name = name
+
+    def describe(self):
+        return 'Validate constraint %s on model %s' % (self.constraint_name, self.model_name)
+
+    def database_forwards(self, app_label, schema_editor, from_state, to_state):
+        model = from_state.apps.get_model(app_label, self.model_name)
+        table_name = schema_editor.quote_name(model._meta.db_table)
+        if self.allow_migrate_model(schema_editor.connection.alias, model):
+            schema_editor.execute("ALTER TABLE %s VALIDATE CONSTRAINT %s" % (
+                table_name, schema_editor.quote_name(self.constraint_name)
+            ))
+
+    def database_backwards(self, app_label, schema_editor, from_state, to_state):
+        pass
+
+    def state_forwards(self, app_label, state):
+        pass
```

Comment:
I think comments on why these are blank would be helpful

```suggestion
    def database_backwards(self, app_label, schema_editor, from_state, to_state):
        # PostgreSQL does not provide a way to make a constraint invalid

    def state_forwards(self, app_label, state):
        # No state needs changing
```